run/deployment-previews/check_status.py

```diff
@@ -0,0 +1,269 @@
+
+TAG_PREFIX = "pr-"
+
+
+def make_tag(pr: str) -> str:
+    return f"{TAG_PREFIX}{pr}"
+
+
+def get_pr(tag: str) -> int:
+    return int(tag.replace(TAG_PREFIX, ""))
+
+
+_default_options = [
+    click.option(
+        "--dry-run",
+        help="Dry-run mode. No tag changes made",
+        default=False,
+        is_flag=True,
+    ),
+]
+
+_cloudrun_options = [
+    click.option("--project-id", required=True, help="Google Cloud Project ID"),
+    click.option(
+        "--region", required=True, help="Google Cloud Region", default="us-central1"
+    ),
+    click.option("--service", required=True, help="Google Cloud Run service name"),
+]
+
+_github_options = [
+    click.option(
+        "--repo-name", required=True, help="GitHub repo name (user/repo, or org/repo)"
+    ),
+    click.option(
+        "--ghtoken-secretname",
+        default="github_token",
+        help="Google Secret Manager secret name",
+    ),
+]
+
+
+def add_options(options: List[dict]) -> Callable:
+    def _add_options(func: Callable) -> Callable:
+        for option in reversed(options):
+            func = option(func)
+        return func
+
+    return _add_options
+
+
+def error(msg: str, context: str = None) -> None:
+    click.secho(f"Error {context}: ", fg="red", bold=True, nl=False)
+    click.echo(msg)
+    sys.exit(1)
+
+
+def get_service(project_id: str, region: str, service_name: str) -> dict:
+    """Get the Cloud Run service object"""
+    api = discovery.build("run", "v1")
+    fqname = f"projects/{project_id}/locations/{region}/services/{service_name}"
+    try:
+        service = api.projects().locations().services().get(name=fqname).execute()
+    except HttpError as e:
+        error(re.search('"(.*)"', str(e)).group(0), context="finding service")
+    return service
+
+
+def get_revision_url(service_obj: dict, tag: str) -> str:
+    """Get the revision URL for the tag specified on the service"""
+    for revision in service_obj["status"]["traffic"]:
+        if revision.get("tag", None) == tag:
+            return revision["url"]
+
+    error(
+        f"Tag on service {service_obj['metadata']['name']} does not exist.",
+        context=f"finding revision tagged {tag}",
+    )
+
+
+def get_revision_tags(service: dict) -> List[str]:
+    """
+    Get all tags associated to a service
+    """
+    revs = []
+
+    for revision in service["status"]["traffic"]:
+        if revision.get("tag", None):
+            revs.append(revision)
+    return revs
+
+
+def github_token(project_id: str, ghtoken_secretname: str) -> str:
+    """Retrieve GitHub developer token from Secret Manager"""
+    client = secretmanager.SecretManagerServiceClient()
+    name = f"projects/{project_id}/secrets/{ghtoken_secretname}/versions/latest"
+    try:
+        response = client.access_secret_version(name=name)
+    except NotFound as e:
+        error(e, context=f"finding secret {ghtoken_secretname}")
+
+    github_token = response.payload.data.decode("UTF-8")
```

Comment:
add comment about decoding since you encoded it when adding to the secret manager